src/core/cgroup.c

```diff
@@ -2584,6 +2584,73 @@ int unit_get_ip_accounting(
         return r;
 }
 
+/* The below is borrowed from cgtop/cgtop.c */
+int unit_get_io_accounting(
+                Unit *u,
+                CGroupIOAccountingMetric metric,
+                uint64_t *ret) {
+
+        int r;
+        _cleanup_free_ char *p = NULL;
+        _cleanup_fclose_ FILE *f = NULL;
+        uint64_t wr = 0, rd = 0;
+
+        assert(u);
+        assert(metric >= 0);
+        assert(metric < _CGROUP_IO_ACCOUNTING_METRIC_MAX);
+        assert(ret);
+
+        if (!UNIT_CGROUP_BOOL(u, io_accounting))
+                return -ENODATA;
+
+        if (!u->cgroup_path)
+                return -ENODATA;
+
+        /* The root cgroup doesn't expose this information */
+        if (unit_has_root_cgroup(u))
+                return -ENODATA;
+
+        r = cg_get_path("io", u->cgroup_path, "io.stat", &p);
+        if (r < 0)
+                return -ENODATA;
+
+        f = fopen(p, "re");
+        if (!f)
+                return -ENODATA;
```

Comment:
Unlike the original, this returns a fake error value. The error can be anything like -EPERM, -ENOMEM, -ENOFILE, etc. Do the same as original, and translate -ENOENT into -ENODATA, but propagate any other error.